### PR TITLE
[13.0][FIX] ddmrp_product_replace: "considered as demand" products, also for qualified demand + ability to copy packagings

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
* This way product replaced by a given buffer can affect the net flow position of the replacing buffer on the demand side as well.

* Ability to copy packagings.

@ForgeFlow

### CODE REVIEW

**Review by TDu — APPROVED** [review]

File: ddmrp_product_replace/models/stock_buffer.py [hunk 1]
```diff
@@ -149,6 +149,22 @@ def _search_stock_moves_incoming_domain(self, outside_dlt=False):
...
         return domain
 
+    def _search_stock_moves_qualified_demand_domain(self):
+        domain = super()._search_stock_moves_qualified_demand_domain()
+        if not self.demand_product_ids:
+            return domain
+        index_replace = False
+        for n, clause in enumerate(domain):
+            if isinstance(clause, tuple) and clause[0] == "product_id":
+                index_replace = n
```
Comment (simahawk) [inline ddmrp_product_replace/models/stock_buffer.py]:
```suggestion
                index_replace = n
                break
```

Comment (simahawk) [reply ddmrp_product_replace/models/stock_buffer.py]: AFAIS you are replacing a leaf here.
Normally I do something like this which seems more readable:
```py
domain = [x for x in domain if x[0] != "product_id"] +  [("product_id", "in", self.demand_product_ids.ids)]
```

**Review by simahawk — APPROVED** [review]

File: ddmrp_product_replace/models/stock_buffer.py [hunk 2]
```diff
@@ -149,6 +149,22 @@ def _search_stock_moves_incoming_domain(self, outside_dlt=False):
...
+        if isinstance(index_replace, int):
```
Comment (mmequignon) [inline ddmrp_product_replace/models/stock_buffer.py]:
```suggestion
        if index_replace is not False:
```

**Review by mmequignon — APPROVED** [review]
technical review only

File: ddmrp_product_replace/models/stock_buffer.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply ddmrp_product_replace/models/stock_buffer.py]: Yes, I agree. Added :+1:

**Pull request merged**